documentation/boltspec_reference.md

```diff
@@ -0,0 +1,1121 @@
+# BoltSpec reference
```

Comment:
Is there a reason not to include `execute_no_plan` and `execute_any_plan` here? [This PR](https://github.com/puppetlabs/bolt/commit/6fdbaf25f3d94d54cfff0e6605c6154e0365618b#diff-f84e1d7c038c6793ef936447c02fa9c43dbf03afe70213d0b3f7eb75dc71509bR176) includes an example and documentation.

Reply:
I did have that originally, but removed it cause it was just a duplicate of what was in `testing_plans.md`. I can add it back.

Reply:
Here it is in `testing_plans.md` btw: https://github.com/puppetlabs/bolt/pull/2649/files#diff-2c26325655047f2d520c02a0f4cb9101097e114b65d6db4ed0c551940b502e42R444

Reply:
Yeah, I think it's best to have them in the reference as well for completeness. And then it probably wouldn't hurt to link to testing_plans from each